packages/editor-ui/src/views/NodeView.v2.vue

```diff
@@ -407,7 +382,10 @@ type AddNodeOptions = {
 	isAutoAdd?: boolean;
 };
 
-async function addNode(node: AddNodeData, options: AddNodeOptions): Promise<INodeUi | undefined> {
+async function onNodeCreate(
+	node: AddNodeData,
+	options: AddNodeOptions = {},
+): Promise<INodeUi | undefined> {
```

Comment:
To make this type more reliable, we should, perhaps, instead of returning `undefined`, throw an error & show a notification to the user that the connection isn't allowed or is incompatible. Otherwise always return `INodeUi`

Reply:
Great point! Done.